numba/core/debuginfo.py

```diff
@@ -422,8 +422,12 @@ def _di_subroutine_type(self, line, function, argmap):
 
         for idx, (name, nbtype) in enumerate(argmap.items()):
             name = name.replace('.', '$')    # for gdb to work correctly
-            datamodel = self.cgctx.data_model_manager[nbtype]
-            lltype = self.cgctx.get_value_type(nbtype)
+            use_ty = nbtype
+            if isinstance(nbtype, types.Omitted):
+                resolver = self.cgctx.typing_context.resolve_value_type
+                use_ty = resolver(nbtype.value)
+            datamodel = self.cgctx.data_model_manager[use_ty]
+            lltype = self.cgctx.get_value_type(use_ty)
```

Comment:
Note to self, it's probably this part patch from #7729 that's causing the build to fail.